### DISCUSSION TITLE
Unwrap_ll.grd files have different size respect corr_ll.grd

### DISCUSSION THREAD

**Discussion by marcontk** [gmtsar/gmtsar#685] [Q&A]
Hello,
I am using gmtsar with mintpy with a full F2 subswath, but Mintpy tells me that the sizes in corr_ll.grd and unwrap_ll.grd are different.

I am using :

proj_ra2ll.csh trans.dat corr.grd corr_ll.grd

proj_ra2ll.csh trans.dat unwrap.grd unwrap_ll.grd
 
corr_ll.grd are 4050 x 2690 in size but the unwrap_ll.grd vary as seen below.

***
WARNING: NOT all input unwrapped interferograms have the same row/column number!
The most common size is: (4050, 2690)

The following dates have different size:
	220904-221022 (3970, 2690)
	220928-221010 (4020, 2690)
	220928-221022 (4020, 2690)
	221010-221022 (3950, 2690)
	221010-221103 (4020, 2690)
	221022-221103 (4020, 2690)
***

How can I make it so that they keep the size when going from radar to geographic coordinates while keeping the sizes?

**Comment by Xiaohua-Eric-Xu** [comment]
Hi @marcontk, this could happen and is one of the reason we are doing time-series in radar coordinates. The acquisitions of SAR satellites does not always cover the same area and so are the interferograms (common area between the two). In radar coordinates, we usually just fill that with NaNs (corr.grd are zeros unless it has no data), but after geocoding, we cropped out all NaNs so after conversion to kml, the plotted png files matches the bounds exactly (nice for viewing). But that may cause problem if you would like to use the geocoded products directly for time-series analysis, as they may covers slightly different area on Earth. One easy way to fix these is to run a `gmt grdinfo */unwrap_ll.grd -C | awk '{print $2,$3,$4,$5}' > tmp.dat` and then a `gmt gmtinfo tmp.dat` so you know the min/max on the bounds of these grids. Then simply cut or sample all of them to the same area before using mintpy with `gmt grdcut/grdsample`. Another way of course is to set up the bounds when running geocoding. You'll need to duplicate a proje_ra2ll.csh and then modify line 74 to hardcode the bounds in the script. Then use the modified script for doing geocoding should produce grids with exactly the same bounds.

**Reply by marcontk** [reply]
Excellent, thank you very much for your help and for clarifying the situation, I will try the two ways you indicate.

**Reply by marcontk** [reply]
I managed to solve grids size, but I have another problem, when I convert topo_ra.grd to lon/lat using proj_ra2ll.csh, then grd2kml shows me the inverted DEM. What is the correct way to get a dem_ll.grd with the same bounds as corr_ll.grd ?

**Reply by Xiaohua-Eric-Xu** [reply]
Hi @marcontk , the topo_ra.grd is upside down so as to match the data storage in the SLC data. It is also related to how GMT/netcdf stores the data. p.s., you don't have to project topo_ra.grd to geocoordinates when you already have a dem.grd (in geocoordinates). Also topo_ra.grd is elevation above a reference radius in the PRM, not the actual elevation above ellipsoid or geoid.

**Reply by marcontk** [reply]
Ok I understand. I thought I could shorten path with topo_ra.grd. MintPy requires the same size for all pairs: unwrapped phase, coherence and DEM in geocoordinates. Maybe making a mask to cut inside the DEM, since my dem.grd has bigger dimensions than the subswath.

**Reply by Xiaohua-Eric-Xu** [reply]
You could run `gmt grdsample dem.grd -Rcorr_ll.grd -Gdem_resamp.grd`, and then use dem_resamp.grd as the DEM for whatever correction in mintpy.

**Reply by marcontk** [reply]
I will try that command. Thank you very much porfesor, you have been a great help.
